src/nodes.coffee

```diff
@@ -1186,7 +1192,7 @@ exports.Obj = class Obj extends Base
     for prop in @properties
       prop = prop.value if prop instanceof Assign and prop.context is 'object'
       prop = prop.unwrapAll()
-      prop.eachName iterator
+      prop.eachName iterator if prop.eachName?
```

Comment:
Without it, this test fails:
```
  #4130: unassignable in destructured param
  AssertionError: Expected [stdin]:2:12: error: keyword ‘null’ can’t be assigned
  @param : null
           ^^^^ to equal TypeError: prop.eachName is not a function
```

I guess there are checks for assignability that happen _after_ `eachName` gets run. So if we want to lose this existence check, we’ll have to make sure that validation happens before this.